server/lib/src/idm/oauth2.rs

```diff
@@ -1342,14 +1342,20 @@ impl<'a> IdmServerProxyReadTransaction<'a> {
             .map(str::to_string)
             .collect();
         if req_scopes.is_empty() {
-            admin_error!("Invalid oauth2 request - must contain at least one requested scope");
+            admin_error!("Invalid OAuth2 request - must contain at least one requested scope");
             return Err(Oauth2Error::InvalidRequest);
         }
 
         // Check the scopes by our scope regex validation rules.
         if !req_scopes.iter().all(|s| OAUTHSCOPE_RE.is_match(s)) {
             admin_error!(
-                "Invalid oauth2 request - requested scopes failed to pass validation rules"
+                "Invalid OAuth2 request - requested scopes ({}) failed to pass validation rules - all need to match the regex {}",
+                req_scopes
```

Comment:
Might be clearer to put this over two lines with "let failed_scopes = ..." then print that.